contracts/base/PeripheryPayments.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {ERC20} from "solmate/tokens/ERC20.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {SafeTransferLib} from "solmate/utils/SafeTransferLib.sol";
+import {IPeripheryPayments} from "../interfaces/IPeripheryPayments.sol";
+
+using CurrencyLibrary for Currency;
+using SafeTransferLib for address;
+using SafeTransferLib for ERC20;
+
+error InsufficientToken();
```

Comment:
I'd personally define these inside the contract too? But dont feel as strongly about that... maybe @marktoda has an opinion?

Reply:
maybe this was even copied from before